### DISCUSSION TITLE
meshopt_simplify() returns 0 indices

### DISCUSSION THREAD

**Discussion by daniilvinn** [zeux/meshoptimizer#689] [Q&A]
Hello! I have a question regarding `meshopt_simplify()`.

When the simplifier is unable to generate an LOD, it can return an index buffer with exactly the same index count as the input index buffer.

But sometimes it returns an index buffer with 0 indices, even though input index buffer contains data.

Why does it happen? Why `meshopt_simplify()` returns an index buffer with 0 indices?

**Comment by zeux** [comment]
Do you have an example input? This could happen in theory but the meshes would need to be fairly odd, eg all triangles are degenerate. Of course if you ask for a very small target number like 0 or 1 you might get 0.

**Reply by daniilvinn** [reply]
Hello! I am not sure I can reproduce it, because I literally lost the input. I am developing Nanite-like renderer and tweaking some values (partitioning settings, target error etc.) led to such kind of results. 
I just wonder what could *theoretically* be the reason for such output, or degenerate triangles are the only possible reason?

**Reply by zeux** [reply]
Well, there's some other cases probably, for example if you ask to simplify a very skinny quad (2 triangles) to 1 triangle, there's a chance that the quad diagonal will be picked for the collapse instead of an edge, and instead of 1 triangle you'll get 0. This in general should be uncommon but there might be situations where this happens. Obviously if your target is 0 triangles you'll also get 0 :)

**Reply by daniilvinn** [reply]
Thanks for explanation!
